src/Routing/CreateUIRouteProvider.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\entity\Routing\CreateUIRouteProvider.
+ */
+
+namespace Drupal\entity\Routing;
+
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Entity\Routing\EntityRouteProviderInterface;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Additional common routes needed for a CRUD UI.
+ *
+ * - add bundle add overview page (like /node/add)
+ * - a generic bundle specific entity add form (like /node/add/page)
+ */
+class CreateUIRouteProvider implements EntityRouteProviderInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRoutes(EntityTypeInterface $entity_type) {
+    $routes = new RouteCollection();
+
+    if ($route = $this->collectionRoute($entity_type)) {
+      $routes->add('entity.' . $entity_type->id() . '.collection', $route);
+    }
+    if ($route = $this->addPageRoute($entity_type)) {
+      $routes->add('entity.' . $entity_type->id() . '.add_page', $route);
+    }
+    if ($route = $this->addFormRoute($entity_type)) {
+      $routes->add('entity.' . $entity_type->id() . '.add_form', $route);
+    }
+
+    return $routes;
+  }
+
+  /**
+   * Returns the collection route.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
+   *
+   * @return \Symfony\Component\Routing\Route|null
+   */
+  protected function collectionRoute(EntityTypeInterface $entity_type) {
+    if ($entity_type->hasLinkTemplate('collection')) {
+      $route = new Route($entity_type->getLinkTemplate('collection'));
+      $route->setDefault('_title', $entity_type->getLabel() . ' content');
```

Comment:
well we can start extending the entity type classes also, but that should be a separate issue/PR imhoi